frontend/src/views/SubmissionView.vue

```diff
@@ -30,7 +34,18 @@ function onCancel() {
   editable.value = false;
 }
 
-async function onSubmit(data: any) {
+function onPermitDelete(data: any) {
+  permits.value = permits.value.filter((x: Permit) => x.permitId !== data.permitId);
```

Comment:
this sure feels simpler than using the store. and permits.value is handled with reactivity.